### ISSUE TITLE
TLS certificate authentication (client-side on prom-migrator, server-side on promscale)

### ISSUE THREAD

**Issue by andrejpodzimek** [timescale/promscale#572]
My Prometheus instance uses the following `web-config.yml`:

```
tls_server_config:
  cert_file: /etc/prometheus/fullchain.pem
  key_file: /etc/prometheus/privkey.pem
  client_auth_type: RequireAndVerifyClientCert
  client_ca_file: /var/ca/ca-bundle.pem
```

There are public LetsEncrypt certificates for server identification, but locally issued certificates from my authority are required for client authentication. Both Grafana and a web browser (e.g. Google Chrome) can connect to Prometheus and authenticate just fine.

With `prom-migrator`, however, I’m getting this error, because there is no way to specify a client certificate:

```
level=error ts=2021-04-08T21:37:32.736Z caller=main.go:115 msg="running reader: remote-run run: executing client-read: read-channels: error sending request: Post \"https://<<<my domain>>>:9090/api/v1/read\": remote error: tls: bad certificate"
```

Symmetrically, on `promscale` itself, there seems to be no way to configure server-side client certificate validation, i.e., there is `-tls-cert-file` and `-tls-key-file`, which works fine (without client certificate validation), but one cannot specify anything like `-tls-ca-file` / `-tls-ca-bundle` or client certificate validation [options](https://golang.org/pkg/crypto/tls/#ClientAuthType).

**Comment by Harkishen-Singh** [comment]
Thank you for opening this. Yes, currently Prom-migrator does not support certificates. I still implement this in the next version. 

Reason: prom-migrator is a new project and so the priority was to get the main functionality right and reliable and then focus on minute things. Now I think, it's a good time to focus on it.

**Comment by Harkishen-Singh** [comment]
@andrejpodzimek we just released `0.0.1` with this fix. You can read more changes [here](https://github.com/timescale/promscale/releases/tag/0.4.0). Let me know if the problem persists.
